pkg/strategy/grid2/recover.go

```diff
@@ -0,0 +1,305 @@
+package grid2
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+	"github.com/pkg/errors"
+)
+
+func (s *Strategy) recoverByScanningTrades(ctx context.Context, session *bbgo.ExchangeSession) error {
+	historyService, implemented := session.Exchange.(types.ExchangeTradeHistoryService)
+	// if the exchange doesn't support ExchangeTradeHistoryService, do not run recover
+	if !implemented {
+		s.logger.Warn("ExchangeTradeHistoryService is not implemented, can not recover grid")
+		return nil
+	}
+
+	s.logger.Infof("recover grid with group id: %d", s.OrderGroupID)
+
+	openOrders, err := session.Exchange.QueryOpenOrders(ctx, s.Symbol)
+	if err != nil {
+		return errors.Wrapf(err, "check if strategy exists error when query open orders")
+	}
+
+	s.logger.Infof("found %d open orders left on the %s order book", len(openOrders), s.Symbol)
+
+	// filter out the order with the group id belongs to this grid
+	var openOrdersOnGrid []types.Order
```

Comment:
can be just `openOrders`? since you don't have the other `openOrders` variable?

Reply:
We already have openOrders at line 25. We filter it by group order id here.